src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2001,6 +2043,22 @@ private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, i
             }
         }
 
+        private static unsafe void ThreadSafeCopy(char* sourcePtr, Span<char> destination, int destinationIndex, int count)
```

Comment:
After spending more time on it, I realized that the ThreadSafeCopy overload was only used by CopyTo, so I don't actually have to impact the other codepaths